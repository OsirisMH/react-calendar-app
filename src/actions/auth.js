import Swal from "sweetalert2";
import { NotificationManager } from 'react-notifications';

import { fetchConToken, fetchSinToken } from "../helpers/fetch";
import { eventsClear } from "./events";
import { types } from "../types/types";

export const startLogin = ( email, password ) => {
    return async ( dispatch ) => {
        try {
            const resp = await fetchSinToken( 'auth', { email, password }, 'POST' );
            const body = await resp.json();
            
            if ( body.ok ) {
                localStorage.setItem( 'token', body.token );
                localStorage.setItem('token-init-date', new Date().getTime() );
                
                dispatch( login({
                    uid: body.uid,
                    name: body.name
                }));
            }
            else {
                Swal.fire('Error', body.msg, 'error');
            }
        }
        catch {

        }
    }
};

export const startRegister = ( name, email, password ) => {
    return async ( dispatch ) => {
        try {
            const resp = await fetchSinToken( 'auth/new', { name, email, password }, 'POST' );
            const body = await resp.json();
            
            if ( body.ok ) {
                localStorage.setItem( 'token', body.token );
                localStorage.setItem('token-init-date', new Date().getTime() );
                
                dispatch( login({
                    uid: body.uid,
                    name: body.name
                }));
            }
            else {
                Swal.fire('Error', body.msg, 'error');
            }
        }
        catch(error) {
            console.log(error);
        }
    }
};

export const startChecking = () => {
    return async ( dispatch ) => {
        try {
            const resp = await fetchConToken( 'auth/renew' );
            const body = await resp.json();
            
            if ( body.ok ) {
                localStorage.setItem( 'token', body.token );
                localStorage.setItem('token-init-date', new Date().getTime() );
                dispatch( login({
                    uid: body.uid,
                    name: body.name
                }));
            }
            else {
                dispatch( checkingFinish() );
                if ( body.msg === 'Token no vÃ¡lido'){
                    NotificationManager.error( 'Session expired', 'Error' );
                }
            }
        }
        catch {
            dispatch( checkingFinish() );
        }
    }
};

const login = ( user ) => ({
    type: types.authLogin,
    payload: user
});

const checkingFinish = () => ({ type: types.authCheckingFinish });

export const startLogout = () => {
    return ( dispatch ) => {
        localStorage.clear();
        dispatch( eventsClear() );
        dispatch( logout() );
    };
};

const logout = () => ({ type: types.authLogout });