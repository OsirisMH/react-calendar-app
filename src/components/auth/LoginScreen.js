import React, { useState } from 'react';
import Swal from 'sweetalert2';

import { useDispatch } from 'react-redux';
import { useForm } from '../../hooks/useForm';

import { startLogin, startRegister } from '../../actions/auth';

import './login.css';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [isLoginIn, setIsLoginIn] = useState( true );
    
    const [ formLoginValues, handleLoginInputChange ] = useForm({
        loginEmail: '',
        loginPassword: ''
    });

    const [ formRegisterValues, handleRegisterInputChange ] = useForm({
        registerName: '',
        registerEmail: '',
        registerPassword: '',
        registerPasswordConfirm: ''
    });

    const {loginEmail, loginPassword } = formLoginValues;

    const { registerName, registerEmail, registerPassword, registerPasswordConfirm } = formRegisterValues;

    const handleClickChange = () => {
        setIsLoginIn( !isLoginIn );  
    };

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(startLogin(loginEmail, loginPassword));
    };

    const handleRegister = (e) => {
        e.preventDefault();

        if ( registerPassword !== registerPasswordConfirm ){
            return Swal.fire('Error', 'Las contrase単as deben ser iguales', 'error');
        }
        else {
            dispatch(startRegister( registerName, registerEmail, registerPassword ));
        }
    };

    return (
        <div className="container login-container">
            <div className="row">
                <div className={`col-md-6 login-form-1 mx-auto ${ !isLoginIn && 'd-none' }`}>
                    <h3>Ingreso</h3>
                    <form onSubmit={ handleLogin }>
                        <div className="form-group">
                            <input 
                                type="text"
                                className="form-control"
                                placeholder="Correo"
                                name="loginEmail"
                                value={ loginEmail }
                                onChange={ handleLoginInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contrase単a"
                                name="loginPassword"
                                value={ loginPassword }
                                onChange={ handleLoginInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input 
                                type="submit"
                                className="btnSubmit"
                                value="Login" 
                            />
                        </div>
                    </form>
                </div>
                <div className={ `col-md-6 login-form-2 mx-auto ${ isLoginIn && 'd-none' }` }>
                    <h3>Registro</h3>
                    <form onSubmit={ handleRegister }>
                        <div className="form-group">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Nombre"
                                name="registerName"
                                value={ registerName }
                                onChange={ handleRegisterInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name="registerEmail"
                                value={ registerEmail }
                                onChange={ handleRegisterInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contrase単a"
                                name="registerPassword"
                                value={ registerPassword }
                                onChange={ handleRegisterInputChange }
                            />
                        </div>

                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la contrase単a" 
                                name="registerPasswordConfirm"
                                value={ registerPasswordConfirm }
                                onChange={ handleRegisterInputChange }
                            />
                        </div>

                        <div className="form-group">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Crear cuenta" />
                        </div>
                    </form>
                </div>
            </div>
            <button
                className={ `btn ${ isLoginIn ? 'btn-primary' : 'btn-outline-primary' } btn-block mt-2 mx-auto w-25` }
                onClick={ handleClickChange }
            >
                { (isLoginIn) ? "Register" : "Login" }
            </button>
        </div>
    )
}