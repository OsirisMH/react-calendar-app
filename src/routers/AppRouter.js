import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
    BrowserRouter as Router,
    Switch,
    Redirect
} from "react-router-dom";
import ReactLoading from 'react-loading';
import NotificationContainer from 'react-notifications/lib/NotificationContainer';

import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

import { startChecking } from '../actions/auth';

import '../styles.css';
import 'react-notifications/lib/notifications.css';

export const AppRouter = () => {
    
    const dispatch = useDispatch();
    const { checking, uid } = useSelector(state => state.auth)
    
    useEffect(() => {
        dispatch( startChecking() );
    }, [dispatch]);

    if ( checking ) {
        return (
            <div className="loadingContainer">
                <h5>Loading...</h5>
                <ReactLoading
                    type="spin"
                    color="#0062cc"
                    height={60}
                    width={60}
                    className="loading"
                />
            </div> 
        );
    }

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute
                        exact
                        path="/login"
                        component={ LoginScreen }
                        isAuthenticated={ !!uid }
                    />

                    <PrivateRoute
                        exact
                        path="/"
                        component={ CalendarScreen }
                        isAuthenticated={ !!uid }
                    />
                    <Redirect to="/" />
                </Switch>
                <NotificationContainer />
            </div>
        </Router>
    )
};
