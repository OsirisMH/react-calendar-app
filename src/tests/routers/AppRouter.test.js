import { mount } from "enzyme";
import { Provider } from "react-redux";

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { AppRouter } from "../../routers/AppRouter";

import * as authModule from '../../actions/auth';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe('Pruebas en <AppRouter />', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('Debe de mostrar el loading...', () => {
        const initState = { auth: { checking: true } };
        const store = mockStore( initState );
        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe de mostrar la ruta pÃºblica', () => {
        const initState = { auth: { checking: false, uid: null } };
        const store = mockStore( initState );
        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.login-container').exists() ).toBe( true );
    });

    test('Debe de mostrar la ruta privada', () => {
        const initState = { ui: { modalOpen: false }, calendar: { events: [] } ,auth: { checking: false, uid: '123', name: 'test' } };
        const store = mockStore( initState );

        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.calendar-screen').exists() ).toBe( true );
    });

    test('Debe llamarse startChecking', async () => {
        authModule.startChecking = jest.fn();
        
        const initState = { auth: {checking: true} };
        const store = mockStore( initState );
        store.dispatch = jest.fn();

        mount( 
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect( authModule.startChecking ).toHaveBeenCalled();
    })
    
});
