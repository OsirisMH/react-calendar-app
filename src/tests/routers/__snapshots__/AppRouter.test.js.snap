// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el loading... 1`] = `
<div
  className="loadingContainer"
>
  <h5>
    Loading...
  </h5>
  <div
    className="loading"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path opacity=\\".25\\" d=\\"M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4\\"/>
  <path d=\\"M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z\\">
    <animateTransform attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 16 16\\" to=\\"360 16 16\\" dur=\\"0.8s\\" repeatCount=\\"indefinite\\" />
  </path>
</svg>
",
      }
    }
    style={
      Object {
        "fill": "#0062cc",
        "height": 60,
        "width": 60,
      }
    }
  />
</div>
`;

exports[`Pruebas en <AppRouter /> Debe de mostrar la ruta privada 1`] = `
<div>
  <div
    className="calendar-screen"
  >
    <div
      className="navbar navbar-dark bg-dark mb-4"
    >
      <span
        className="navbar-brand"
      >
        test
      </span>
      <button
        className="btn btn-outline-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt pr-2"
        />
        <span>
          Salir
        </span>
      </button>
    </div>
    <div
      className="rbc-calendar"
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Hoy
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            &lt;
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            &gt;
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          julio 2021
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Mes
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Semana
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Día
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        className="rbc-month-view"
        role="table"
      >
        <div
          className="rbc-row rbc-month-header"
          role="row"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              dom.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              lun.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              mar.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              mié.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              jue.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              vie.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              sáb.
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  10
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  12
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  17
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  19
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  24
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  31
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </button>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <div />
  </div>
</div>
`;

exports[`Pruebas en <AppRouter /> Debe de mostrar la ruta pública 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1 mx-auto false"
      >
        <h3>
          Ingreso
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginEmail"
              onChange={[Function]}
              placeholder="Correo"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginPassword"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2 mx-auto d-none"
      >
        <h3>
          Registro
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerName"
              onChange={[Function]}
              placeholder="Nombre"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerEmail"
              onChange={[Function]}
              placeholder="Correo"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPassword"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPasswordConfirm"
              onChange={[Function]}
              placeholder="Repita la contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
    <button
      className="btn btn-primary btn-block mt-2 mx-auto w-25"
      onClick={[Function]}
    >
      Register
    </button>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <div />
  </div>
</div>
`;
